rust/core/src/automatic_branching.rs

```diff
@@ -357,73 +387,234 @@ mod tests {
+
+        assert_eq!(hash_0, hash_1);
+    }
+
+    #[tokio::test]
+    async fn test_derive_snapshot_hash_change_snapshot_definition() {
+        let mut fs = init_to_file_system();
         fs.files.insert(
-            "models/schema_2.yaml".to_string(),
+            "models/employees_snapshot_config.yaml".to_string(),
             File {
-                name: "models/schema_2.yaml".to_string(),
+                name: "models/employees_snapshot_config.yaml".to_string(),
                 contents: prost::bytes::Bytes::from(
                     r#"
-                        models: 
-                            - name: shifts_summary_2
-                              description: test description for model
+                    snapshots:
+                    - name: employees_snapshot
+                      unique_key: id
+                      strategy:
+                        timestamp:
+                          updated_at: update_at
                     "#
                     .to_string(),
                 ),
             },
         );
+        fs.files.insert(
+            "models/employees_snapshot.snapshot.sql".to_string(),
+            File {
+                name: "models/employees_snapshot.snapshot.sql".to_string(),
+                contents: prost::bytes::Bytes::from(
+                    r#"
+                        SELECT
+                            *
+                        FROM
+                            q.raw_employees
+                    "#
+                    .to_string(),
+                ),
+            },
+        
+        );
 
-        let hash_1 = get_hash_derive_model_hash(&fs).await;
+        let hash_0 = get_hash_derive_model_hash(&fs, Some("employees_snapshot")).await;
 
-        assert_eq!(hash_0, hash_1);
+        let path_of_interest = "models/employees_snapshot.snapshot.sql";
+        let mut raw_file = fs.read_file(path_of_interest).await.unwrap();
+        let mut contents = String::new();
+        raw_file.read_to_string(&mut contents).await.unwrap();
+        let contents = contents.replace(' ', "  ");
+        fs.files.insert(
+            path_of_interest.to_string(),
+            File {
+                name: path_of_interest.to_string(),
+                contents: prost::bytes::Bytes::from(contents),
+            },
+        );
+        let hash_1 = get_hash_derive_model_hash(&fs, Some("employees_snapshot")).await;
+
+        assert_ne!(hash_0, hash_1);
     }
 
+    #[tokio::test]
+    async fn test_derive_snapshot_hash_change_snapshot_configuration() {
+        let mut fs = init_to_file_system();
+        fs.files.insert(
+            "models/employees_snapshot_config.yaml".to_string(),
+            File {
+                name: "models/employees_snapshot_config.yaml".to_string(),
+                contents: prost::bytes::Bytes::from(
+                    r#"
+                    snapshots:
+                    - name: employees_snapshot
+                      unique_key: id
```

Comment:
nit: ordering here is quite jarring because you are splitting a set of tests that belong together and are testing the same thing.